runtime/domain.c

```diff
@@ -1732,6 +1754,12 @@ static void domain_terminate (void)
   caml_plat_signal(&domain_self->domain_cond);
   caml_plat_unlock(&domain_self->domain_lock);
 
+#ifndef _WIN32
```

Comment:
I think this should be `#ifdef POSIX_SIGNALS`.

Also, are we doing this too late? A signal can still arrive after we've reached a stage where there's no more polling points on this domain. The issue is we'll set `young_limit` but then never check it, leading to a delayed signal (eventually another domain will probably do a GC or terminate).

Reply:
> I think this should be #ifdef POSIX_SIGNALS.

I was going by the other uses of `pthread_sigmask` in `domain.c` which are protected by `#ifndef _WIN32`. If this should be changes, perhaps the other ones too. CC @dra27. 

> Also, are we doing this too late? A signal can still arrive after we've reached a stage where there's no more polling points on this domain. The issue is we'll set young_limit but then never check it, leading to a delayed signal (eventually another domain will probably do a GC or terminate).

If we move this to be the first thing that we do in `domain_terminate`, is that any better? No OCaml callbacks run in `domain_terminate`.

Reply:
`POSIX_SIGNALS` is the better one - I should have used that when I did the others!

Reply:
Thanks. Let me use `POSIX_SIGNALS` for this one. Will you make a PR for the existing ones on trunk @dra27?

Reply:
See #11219